security/login_link.rst

```diff
@@ -0,0 +1,510 @@
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/security.yaml
+        security:
+            firewalls:
+                main:
+                    login_link:
+                        check_route: login_check
+                        # only allow the link to be used once
+                        max_uses: 1
+
+                        # optionally, configure the cache pool
+                        #used_link_cache: 'cache.redis'
+
+    .. code-block:: xml
+
+        <!-- config/packages/security.xml -->
+        <?xml version="1.0" encoding="UTF-8"?>
+        <srv:container xmlns="http://symfony.com/schema/dic/security"
+            xmlns:srv="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                https://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/security
+                https://symfony.com/schema/dic/security/security-1.0.xsd">
+
+            <config>
+                <firewall name="main">
+                    <!-- max-uses: only allow the link to be used once
+                         used-link-cache: optionally, configure the cache pool -->
+                    <login-link check-route="login_check"
+                        max-uses="1"
+                        used-link-cache="cache.redis"
+                    />
+                </firewall>
+            </config>
+        </srv:container>
+
+    .. code-block:: php
+
+        // config/packages/security.php
+        $container->loadFromExtension('security', [
+            'firewalls' => [
+                'main' => [
+                    'login_link' => [
+                        'check_route' => 'login_check',
+                        // only allow the link to be used once
+                        'max_uses' => 1,
+
+                        // optionally, configure the cache pool
+                        //'used_link_cache' => 'cache.redis',
+                    ],
+                ],
+            ],
+        ]);
+
+.. caution::
+
+    When setting ``max_uses`` to ``1``, you must take extra precautions to
+    make it work as expected. Email providers and browsers often load a
+    preview of the links, meaning that the link is already invalidated by
+    the preview loader.
+
+    In order to solve this issue, add a page between the link the user
+    clicks and the ``check_route`` page. On this page, add a button or
+    action the user can use to login::
+
+        // src/Controller/SecurityController.php
+        namespace App\Controller;
+
+        // ...
+        use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+        class SecurityController extends AbstractController
+        {
+            // ...
+
+            /**
+             * @Route("/process_login", name="process_login")
```

Comment:
`/process_login` feels weird to me, why not `/process-link` ?

- `_`=>`-`
- usage of `link` as inside the `processLink()`